src/HospitalLibrary/Rooms/Repositories/RoomEquipmentRepository.cs

```diff
@@ -10,5 +15,31 @@ public class RoomEquipmentRepository: BaseRepository<RoomEquipment>, IRoomEquipm
         public RoomEquipmentRepository(HospitalDbContext dataContext) : base(dataContext)
         {
         }
+
+        public int GetNumberOfUsedEquipment(int equipmentId)
+        {
+            
+            return _dataContext.EquipmentReallocations
+                .Where(a => a.EquipmentId == equipmentId)
+                .Where(a=> a.state==ReallocationState.PENDING)
+                .Select(a => a.amount).Sum();
+        }
+        public async Task<RoomEquipment> GetEquipmentByRoomAndName(int roomId,string name)
+        {
+            if(!_dataContext.RoomEquipment.Any(a => a.RoomId == roomId && a.Name == name))
+                return null;
+            return await _dataContext.RoomEquipment
```

Comment:
Repositories should not have any business logic. Only thing that repo method should do is generate a query. If else should not be here